keel-web/src/test/kotlin/com/netflix/spinnaker/keel/ApplicationServiceTests.kt

```diff
@@ -0,0 +1,358 @@
+          environments = setOf(
+            ArtifactSummaryInEnvironment(environment = "test", version = "fnord-1.0.1-h1.b1b1b1b", state = "previous", deployedAt = Instant.parse("2020-03-25T03:00:00Z"), replacedAt = Instant.parse("2020-03-25T05:00:00Z"), replacedBy = "fnord-1.0.2-h2.c2c2c2c"),
+            ArtifactSummaryInEnvironment(environment = "staging", version = "fnord-1.0.1-h1.b1b1b1b", state = "current", deployedAt = Instant.parse("2020-03-25T04:00:00Z")),
+            ArtifactSummaryInEnvironment(environment = "production", version = "fnord-1.0.1-h1.b1b1b1b", state = "pending", statefulConstraints = listOf(StatefulConstraintSummary("manual-judgement", NOT_EVALUATED), StatefulConstraintSummary("pipeline", NOT_EVALUATED)))
+          ),
+          build = BuildMetadata(1),
+          git = GitMetadata("b1b1b1b")
+        )
+        val v0 = ArtifactVersionSummary(
+          version = "fnord-1.0.0-h0.a0a0a0a",
+          displayName = "1.0.0",
+          environments = setOf(
+            ArtifactSummaryInEnvironment(environment = "test", version = "fnord-1.0.0-h0.a0a0a0a", state = "previous", deployedAt = Instant.parse("2020-03-25T00:00:00Z"), replacedAt = Instant.parse("2020-03-25T03:00:00Z"), replacedBy = "fnord-1.0.1-h1.b1b1b1b"),
+            ArtifactSummaryInEnvironment(environment = "staging", version = "fnord-1.0.0-h0.a0a0a0a", state = "previous", deployedAt = Instant.parse("2020-03-25T01:00:00Z"), replacedAt = Instant.parse("2020-03-25T04:00:00Z"), replacedBy = "fnord-1.0.1-h1.b1b1b1b"),
+            ArtifactSummaryInEnvironment(environment = "production", version = "fnord-1.0.0-h0.a0a0a0a", state = "current", deployedAt = Instant.parse("2020-03-25T02:00:00Z"), statefulConstraints = listOf(StatefulConstraintSummary("manual-judgement", OVERRIDE_PASS, startedAt = Instant.parse("2020-03-25T00:00:00Z"), judgedBy = "lpollo@acme.com", judgedAt = Instant.parse("2020-03-25T01:30:00Z"), comment = "Aye!"), StatefulConstraintSummary("pipeline", NOT_EVALUATED)))
+          ),
+          build = BuildMetadata(0),
+          git = GitMetadata("a0a0a0a")
+        )
+
+        expect {
+          that(summaries.size).isEqualTo(1)
+          that(summaries.first().versions.find { it.version == "fnord-1.0.3-h3.d3d3d3d" }).isEqualTo(v3)
+          that(summaries.first().versions.find { it.version == "fnord-1.0.2-h2.c2c2c2c" }).isEqualTo(v2)
+          that(summaries.first().versions.find { it.version == "fnord-1.0.1-h1.b1b1b1b" }).isEqualTo(v1)
+          that(summaries.first().versions.find { it.version == "fnord-1.0.0-h0.a0a0a0a" }).isEqualTo(v0)
+        }
+      }
+
+      test("no constraints have been evaluated") {
+        val states = applicationService.getConstraintStatesFor(application, "prod", 10)
+        expectThat(states).isEmpty()
+      }
+
+      test("pending manual judgement") {
+        val judgement = ConstraintState(deliveryConfig.name, "production", "fnord-1.0.2-h2.c2c2c2c", "manual-judgement", PENDING)
```

Comment:
Any reason you can't just use the one stored in the big `before` block for all tests? Then you don't have to check that size is 2 below, which was confusing to me.